src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -649,32 +649,38 @@ private static bool AreEquivalent(StringComparer comparer, INamedTypeSymbol tupl
                 });
         }
 
-        private static SeparatedSyntaxList<TArgumentSyntax> ConvertArguments(SyntaxGenerator generator, SeparatedSyntaxList<TArgumentSyntax> arguments)
+        private SeparatedSyntaxList<TArgumentSyntax> ConvertArguments(SyntaxGenerator generator, SeparatedSyntaxList<TArgumentSyntax> arguments)
             => generator.SeparatedList<TArgumentSyntax>(ConvertArguments(generator, arguments.GetWithSeparators()));
 
-        private static SyntaxNodeOrTokenList ConvertArguments(SyntaxGenerator generator, SyntaxNodeOrTokenList list)
+        private SyntaxNodeOrTokenList ConvertArguments(SyntaxGenerator generator, SyntaxNodeOrTokenList list)
             => new SyntaxNodeOrTokenList(list.Select(v => ConvertArgumentOrToken(generator, v)));
 
-        private static SyntaxNodeOrToken ConvertArgumentOrToken(SyntaxGenerator generator, SyntaxNodeOrToken arg)
+        private SyntaxNodeOrToken ConvertArgumentOrToken(SyntaxGenerator generator, SyntaxNodeOrToken arg)
             => arg.IsToken
                 ? arg
                 : ConvertArgument(generator, (TArgumentSyntax)arg.AsNode());
 
-        private static TArgumentSyntax ConvertArgument(
+        private TArgumentSyntax ConvertArgument(
             SyntaxGenerator generator, TArgumentSyntax argument)
         {
-            // Keep named arguments for literal args.  It helps keep the code self-documenting.
+            // If the original arguments had names then we keep them, but convert the case to match the
+            // the constructor parameters they now refer to. It helps keep the code self-documenting.
             // Remove for complex args as it's most likely just clutter a person doesn't need
             // when instantiating their new type.
             var expr = generator.SyntaxFacts.GetExpressionOfArgument(argument);
             if (expr is TLiteralExpressionSyntax)
             {
-                return argument;
+                var argumentName = generator.SyntaxFacts.GetNameForArgument(argument);
+                var newArgumentName = GetConstructorParameterName(argumentName);
+
+                return GetArgumentWithChangedName(argument, newArgumentName);
             }
 
             return (TArgumentSyntax)generator.Argument(expr).WithTriviaFrom(argument);
         }
 
+        protected abstract TArgumentSyntax GetArgumentWithChangedName(TArgumentSyntax argument, string name);
```

Comment:
nit: please place abstracts together.  ideally at the top, below the constructor.

Reply:
There aren't any others to put this with, but below the constructor is good info, thanks :)